index.js

```diff
@@ -87,7 +87,12 @@ function registerListener(session, options, callback = () => {}) {
 		const errorMessage = options.errorMessage || 'The download of {filename} was interrupted';
 
 		if (options.saveAs) {
-			item.setSaveDialogOptions({defaultPath: filePath});
+			let saveOptions = options.dialogOptions ? options.dialogOptions : {};
+			if (!saveOptions.defaultPath) {
+				saveOptions.defaultPath = filePath;
+			}
```

Comment:
Use object spread instead of this. It's more concise and this is also modifying the user supplied object, which is wrong.

Reply:
Sorry about the reverts, I got confused